pkg/agent/openflow/pipeline.go

```diff
@@ -348,10 +370,24 @@ var (
 	// snatPktMarkRange takes an 8-bit range of pkt_mark to store the ID of
 	// a SNAT IP. The bit range must match SNATIPMarkMask.
 	snatPktMarkRange = binding.Range{0, 7}
+	// servicePktMarkRange takes a 1-bit range of pkt_mark to mark the first packet of NodePort/LoadBalancer packet whose
+	// externalTrafficPolicy is Local and Endpoint is on host network. There is an iptables entry which matches the pkt_mark
+	// and adds a ct_mark to the connection. Note that, this iptables entry only matches the first packet.
+	// There is another iptables entry which matches the ct_mark and sets fw_mark to the matched packets. This iptables
+	// entry is used mark every response packets. There is a ip rule matching the fw_mark and lookuping a target routing
+	// table whose default output interface is Antrea gateway.
+	// Through above steps, the response packets can be forwarded back to Antrea gateway correctly.
+	servicePktMarkRange = binding.Range{8, 8}
 
 	globalVirtualMAC, _ = net.ParseMAC("aa:bb:cc:dd:ee:ff")
 	hairpinIP           = net.ParseIP("169.254.169.252").To4()
 	hairpinIPv6         = net.ParseIP("fc00::aabb:ccdd:eeff").To16()
+
+	// serviceGWHairpinIPv4/serviceGWHairpinIPv6 is used to perform SNAT on Service hairpin packet. The hairpin packet comes
+	// from Antrea gateway and will be output through Antrea gateway. They are also used as the gateway IP address of
+	// host Service routing entry.
+	serviceGWHairpinIPv4 = net.ParseIP("169.254.169.253").To4()
```

Comment:
Yes. We don't need another VIP. `hairpinIP` has no conflict with `serviceGWHairpinIPv4`.

Reply:
> Can we use hairpinIP as the SNAT IP? The IP can represent all hairpin cases that we want to ensure response are transmitted back to OVS.

IMO, if we don't change the existing design of client from pod hairpin Service, we cannot use hairpinIP as the SNAT IP. When dst_ip is hairpinIP, it will be first processed at table serviceHairpinTable as below.  As we can see, the destination IP will be changed, this is not we wanted.
```
table=23, priority=200,ip,nw_dst=169.254.169.252 actions=move:NXM_OF_IP_SRC[]->NXM_OF_IP_DST[],load:0x1->NXM_NX_REG0[18],resubmit(,24)
```
Another way is that we change the SNAT way of client from pod hairpin Service. When a client from pod accesses a Service whose Endpoint is the pod, the SNAT is also implemented through ct_zone. Then we can unify the hairpinIP as one. But this will introduce extra overhead for this case. Existing SNAT is implemented by stateless SNAT(just modifying IP address, without ct)